checkout-core/src/main/java/com/adyen/checkout/core/internal/data/model/JsonUtils.kt

```diff
@@ -95,4 +123,47 @@ object JsonUtils {
             }
         }
     }
+
+    /**
+     * Parses a [JSONArray] to a list of integers.
+     *
+     * @param jsonArray The JSONArray to be read.
+     * @return A [List] of integers, or null if the jsonArray was null.
+     */
+    @JvmStatic
+    @Throws(JSONException::class)
+    fun parseOptIntegerList(jsonArray: JSONArray?): List<Int>? {
+        if (jsonArray == null) {
+            return null
+        }
+        val list: MutableList<Int> = ArrayList()
+        for (i in 0 until jsonArray.length()) {
+            val jsonValue = jsonArray.opt(i)
+            if (jsonValue is Int) {
+                val item = jsonArray.optInt(i)
+                list.add(item)
+            } else {
+                throw JSONException("type is not integer")
+            }
+        }
+        return Collections.unmodifiableList(list)
+    }
+
+    /**
+     * Serializes a List of integers to a [JSONArray].
+     *
+     * @param intList The [List] of integers to be serialized.
+     * @return The populated [JSONArray]. Could be null.
+     */
+    @JvmStatic
+    fun serializeOptIntegerList(intList: List<Int?>?): JSONArray? {
+        if (intList == null) {
+            return null
+        }
+        return JSONArray().apply {
+            intList.filterNotNull().forEach {
```

Comment:
Is there a reason to allow null values in the list? And if we do why should we filter them out? If i am trying to serialize a list of `1, 2, null, 3` i probably want the null to be there